client/lib/menu-data/menu-data.js

```diff
@@ -388,30 +407,76 @@ MenuData.prototype.saveMenu = function( menu, callback ) {
 
 	debug( 'saveMenu', menu );
 
-	this.emit( 'saving' );
-	wpcom
-	.undocumented()
-	.menusUpdate(
-		this.siteID,
-		menu.id,
-		this.interceptSaveForHomepageLink( menu ),
-		( error, data ) => {
-			if ( error ) {
-				this.emit( 'error', i18n.translate( 'There was a problem saving your menu.' ) );
-				debug( 'Error', error, data );
-				return;
-			}
+	const self = this;
+
+	function saveMenu() {
```

Comment:
Why are we defining another function here? Try moving this up a level and passing through the needed arguments.